### ISSUE TITLE
Inconsistent DTensorPolicyWorker logits with TP>1 and bf16

### ISSUE THREAD

**Issue by jveronvialard** [NVIDIA-NeMo/RL#1227]
**Describe the bug**

When using TP>1 and bf16 policy.precision, we observed inconsistent logits computation during the forward pass of DTensorPolicyWorker, impacting RM training.
- Identical rows yield different outputs depending on their position in the microbatch
- Consecutive forward passes produce different outputs

Using fp32 for policy.precision seems to mitigate this issue.
